src/tests/test_sofie_test_parser.py

```diff
@@ -0,0 +1,74 @@
+"answer": "Hvis Mia er 40 år, er Emma 40+16 = <<40+16=56>>56 år.\nSummen af deres alder er 56+40 = <<56+40=96>>96 år\nGennemsnitsalderen for de to er 96/2 = <<96/2=48>>48 år\n#### 48",
+"id_orig": 1277,
+"id_shuffled": 31,
+"question_annotated": "Der er i øjeblikket {age_diff,16} år mellem {name1,Mia} og {name2,Emma}. Hvis {name1,Mia}, wder er yngre end {name2,Emma}, er {age1,40} år gammel, hvad er gennemsnittet af deres alder?\n\n#init:\n- name1, name2 = sample(names, 2)\n- $age_diff = range(5, 30)\n- $age1 = range(15, 75)\n\n#conditions:\n- is_int((2*age1 + age_diff) / 2)\n\n#answer: (2*age1 + age_diff) // 2",
+"answer_annotated": "Hvis {name1} er {age1} år gammel, er {name2} {age1}+{age_diff} = <<{age1}+{age_diff}={age1+age_diff}>>{age1+age_diff} år.\nSummen af deres alder er {age1+age_diff}+{age1} = <<{age1+age_diff}+{age1}={2*age1+age_diff}>>{2*age1+age_diff} år\nGennemsnitsalderen for de to er {2*age1+age_diff}/2 = <<{2*age1+age_diff}/2={(2*age1+age_diff)//2}>>{(2*age1+age_diff)//2} år\n#### {(2*age1+age_diff)//2}"
+}"""
+
+
+@pytest.fixture
+def annotated_question():
+    question = AnnotatedQuestion.from_json(json_data)
+    return question
+
+##### test json input #####
+def test_load_from_json():
+    data = json.loads(json_data)
+    question = AnnotatedQuestion.from_json(json_data)
+    assert question.question == data["question"]
+    assert question.answer == data["answer"]
+
+##### test extraction of default values #####
+def test_default_values_extraction(annotated_question):
+    assert annotated_question.default_values == {'age_diff': '16', 'name1': 'Mia', 'name2': 'Emma', 'age1': '40'}
+    assert "age_diff" in annotated_question.default_values
+    assert "age1" in annotated_question.default_values
+
+##### test apply_init_rules and replace_values #####
+def test_apply_init_rules_and_replace(annotated_question):
+    new_values = annotated_question.apply_init_rules(names, multiple_ice, multi_times, functions)
+    q_text, a_text = annotated_question.replace_values(new_values)
+    assert "{name1}" not in q_text
+
+##### test "calculations" (eval_expressions func) #####
+def test_eval_expressions():
+    fixed_var = {"age1": 40, "age_diff": 16}
+    expr = ("Hvis {name1} er {age1} år gammel, er {name2} {age1}+{age_diff} = <<{age1}+{age_diff}={age1+age_diff}>>{age1+age_diff} år.\n"
+            "Summen af deres alder er {age1+age_diff}+{age1} = <<{age1+age_diff}+{age1}={2*age1+age_diff}>>{2*age1+age_diff} år\n"
+            "Gennemsnitsalderen for de to er {2*age1+age_diff}/2 = <<{2*age1+age_diff}/2={(2*age1+age_diff)//2}>>{(2*age1+age_diff)//2} år\n"
+            "#### {(2*age1+age_diff)//2}")
+    result = eval_expressions(expr, fixed_var)
+    assert "56" in result
+    assert "96" in result
+    assert "48" in result
+
+##### test the generate_question func #####
+def test_generate_question(annotated_question):
+    generated_question = annotated_question.generate_question(names, multiple_ice, multi_times, functions)
```

Comment:
```suggestion
    generated_question = annotated_question.generate_question(names, multiple_ice, multi_times)
```
Should just be a default